backend/internal/server/apis/schedule/games/games.go

```diff
@@ -0,0 +1,286 @@
+	"github.com/jak103/powerplay/internal/server/apis/schedule/pkg/models"
+	"github.com/jak103/powerplay/internal/server/apis/schedule/pkg/optimize"
+	"github.com/jak103/powerplay/internal/server/apis/schedule/pkg/parser"
+	"github.com/jak103/powerplay/internal/utils/log"
+	"github.com/jak103/powerplay/internal/utils/responder"
+)
+
+func init() {
+	apis.RegisterHandler(fiber.MethodPost, "/schedule/games", auth.Authenticated, handleGenerate)
+}
+
+func handleGenerate(c *fiber.Ctx) error {
+	log.Info("Reading body\n")
+	seasonName, numberOfGamesPerTeam, err := readBody(c)
+	if err != nil {
+		log.Error("Error reading body: %v\n", err)
+		return err
+	}
+
+	if seasonName == "" {
+		log.Error("Season file name is empty\n")
+		return responder.BadRequest(c, "Season file name is empty")
+	}
+
+	if numberOfGamesPerTeam == 0 {
+		log.Error("Number of games per team is 0\n")
+		return responder.BadRequest(c, "Number of games per team is 0")
+	}
+
+	log.Info("Reading config file for season\n")
+	seasonConfig, err := parser.SeasonConfig(seasonName)
+	if err != nil {
+		log.Error("Error reading file: %v\n", err)
+		return responder.BadRequest(c, "Error reading file")
+	}
+
+	log.Info("Generating games\n")
+	season, err := generateGames(seasonConfig.Leagues, numberOfGamesPerTeam)
+
+	log.Info("Assigning ice times\n")
+	games, err := assignTimes(seasonConfig.IceTimes, season, numberOfGamesPerTeam)
+	if err != nil {
+		log.Error("Error assigning ice times: %v\n", err)
+		return responder.BadRequest(c, "Error assigning ice times")
+	}
+
+	log.Info("Optimizing schedule\n")
+	optimizeSchedule(games)
+
+	log.Info("Writing csv\n")
+	err = csv.GenerateCsv(games, "schedule.csv")
+	if err != nil {
+		log.Error("Error writing csv: %v\n", err)
+		return responder.InternalServerError(c)
+	}
+
+	//log.Info("Saving to database\n")
+	//err = save.ToDb(c, games)
+	//if err != nil {
+	//	log.Error("Error saving to database: %v\n", err)
+	//	return responder.InternalServerError(c)
+	//}
+
+	return responder.Ok(c, "Schedule generated at schedule.csv and saved to database")
+}
+
+func readBody(c *fiber.Ctx) (string, int, error) {
+	type BodyDto struct {
+		SeasonName           string `json:"seasonName"`
+		NumberOfGamesPerTeam int    `json:"numberOfGamesPerTeam"`
+	}
+	body := c.Body()
+	var bodyDto BodyDto
+	err := json.Unmarshal(body, &bodyDto)
+	if err != nil {
+		return "", 0, responder.BadRequest(c, "Error reading body")
+	}
+
+	return bodyDto.SeasonName, bodyDto.NumberOfGamesPerTeam, nil
+}
+
+func optimizeSchedule(games []models.Game) {
+	if len(games) == 0 {
+		log.Info("No games to optimize")
+		return
+	}
+	log.Info("Pre-optimization analysis")
+	seasonStats, teamStats := analysis.RunTimeAnalysis(games)
+
+	// Need to make sure games are balanced in
+	// - Early / late
+	// - Days between games
+	balanceCount := getBalanceCount(&teamStats)
+	lastBalanceCount := -1
+
+	for count := 0; balanceCount != lastBalanceCount && count < 25; count++ {
```

Comment:
One thing I have seen happen when playing with the scheduler is sometimes when running the optimization multiple times a subsequent run will be less balanced than the previous one. So an alternative to this logic would be to run it multiple times and pick the one that's most balanced

Reply:
That's a good idea. Think we should include it in this PR or do it as another feature?

Reply:
separate feature is OK with me.

Reply:
https://github.com/jak103/powerplay/issues/118